definitions/tools/cnvkit_batch.cwl

```diff
@@ -13,64 +13,89 @@ requirements:
       ramMin: 4000
       tmpdirMin: 10000
     - class: InlineJavascriptRequirement
+arguments: [{ valueFrom: "$((inputs.reference.hasOwnProperty('cnn_file'))? '' : '--normal')" }]
 inputs:
     tumor_bam:
         type: File
         inputBinding:
-            position: 1
-    normal_bam:
-        type: File?
-        inputBinding:
-            position: 2
-            prefix: "--normal"
+            position: -1
     bait_intervals:
         type: File?
         inputBinding:
             position: 3
             prefix: "--targets"
     reference:
-        type:
-            - "null"
-            - string
-            - File
-        secondaryFiles: [.fai, ^.dict]
-        inputBinding:
-            position: 4
-            prefix: "--fasta"
+      type:
+        - type: record
+          name: cnn_file
+          fields:
+            cnn_file:
+              type: File
+              inputBinding:
+                position: 2
+                prefix: "--reference"
+              doc: "Previously generated reference.cnn file"
+        - type: record
+          name: fasta_file
+          fields:
+            fasta_file:
+              type:
+                - string
+                - File
+              inputBinding:
+                position: 2
+                prefix: "--fasta"
+            normal_bam:
+              type: File?
+              inputBinding:
+                position: 1
```

Comment:
Alternatively, I wonder if you could get away with defining a `valueFrom` here that always includes the `--normal` and optionally includes the file when it's present.